### PR TITLE
2759 generic support bundle

### PR DESCRIPTION
**Pull request by keithalucas**
Made the support bundle kit support multiple namespaces and determine what resources are available in each namespace and generate the yaml files for each resources in each namespace.

### CODE REVIEW

File: pkg/utils/namespace.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package utils
+
+import (
+	"io/ioutil"
+	"strings"
+)
+
+// Get the namespace that the Pod is running in.
+func PodNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
```
Comment (yasker) [inline pkg/utils/namespace.go:10]: Normally the pod namespace can be passed in use POD_NAMESPACE environment variable. See https://github.com/longhorn/longhorn/blob/47d38e621e0c52ae9c2a316baf0138a9b09b2fc6/deploy/longhorn.yaml#L739

This file seems to reflect `the default namespace to be used for namespaced API operations`. I am a bit unsure if it's the pod's running namespace. https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod

File: cmd/manager.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ And it also waits for reports from support bundle agents. The reports contain:
 
 func init() {
 	rootCmd.AddCommand(managerCmd)
-	managerCmd.PersistentFlags().StringVar(&sbm.HarvesterNamespace, "harvester-namespace", os.Getenv("HARVESTER_NAMESPACE"), "The Harvester namespace")
+	managerCmd.PersistentFlags().StringVar(&sbm.NamespaceList, "namespaces", os.Getenv("NAMESPACES"), "List of namespaces delimited by :")
```
Comment (yasker) [inline cmd/manager.go:39]: Can we use `,` instead of `:`? Haven't seen the usage of `:` as the delimiter much.

File: pkg/manager/manager.go [hunk 3]
```diff
@@ -135,6 +141,8 @@ func (m *SupportBundleManager) Run() error {
 }
 
 func (m *SupportBundleManager) phaseInit() error {
+	m.Namespaces = strings.Split(m.NamespaceList, ":")
```
Comment (yasker) [inline pkg/manager/manager.go:144]: Use `comma/,` instead.

**Review by yasker — CHANGES_REQUESTED** [review]
LGTM in general. Only minor changes are needed.

File: pkg/utils/namespace.go (hunk 1 above)
Comment (keithalucas) [reply pkg/utils/namespace.go:15]: I updated it use the POD_NAMESPACE environment variable first and fall back to the file if POD_NAMESPACE isn't set.

File: cmd/manager.go (hunk 2 above)
Comment (keithalucas) [reply cmd/manager.go]: I was thinking of the unix PATH, but I updated it to use comma.

File: pkg/manager/manager.go (hunk 3 above)
Comment (keithalucas) [reply pkg/manager/manager.go]: Switched to comma.

**Review by yasker — APPROVED** [review]

**Pull request merged**